app/scripts/modules/core/pipeline/config/triggers/travis/travisTrigger.module.ts

```diff
@@ -0,0 +1,126 @@
+      mastersRefreshing: false,
+      jobsLoaded: false,
+      jobsRefreshing: false,
+    };
+    this.initializeMasters();
+    $scope.$watch(() => trigger.master, () => this.updateJobsList());
+  }
+
+  public refreshMasters(): void {
+    this.viewState.mastersRefreshing = true;
+    this.initializeMasters();
+  };
+
+  public refreshJobs(): void {
+    this.viewState.jobsRefreshing = true;
+    this.updateJobsList();
+  };
+
+  public shouldFilter(): boolean {
+    return this.jobs && this.jobs.length >= this.filterThreshold;
+  }
+
+  private initializeMasters(): void {
+    this.igorService.listMasters(BuildServiceType.Travis).then((masters: string[]) => {
+      this.masters = masters;
+      this.viewState.mastersLoaded = true;
+      this.viewState.mastersRefreshing = false;
+    });
+  }
+
+  private updateJobsList(): void {
+    if (this.trigger && this.trigger.master) {
+      this.viewState.jobsLoaded = false;
+      this.jobs = [];
+      this.igorService.listJobsForMaster(this.trigger.master).then(jobs => {
+        this.viewState.jobsLoaded = true;
+        this.viewState.jobsRefreshing = false;
+        this.jobs = jobs;
+        if (jobs.length && !this.jobs.includes(this.trigger.job)) {
+          this.trigger.job = '';
+        }
+      });
+    }
+  }
+}
+
+export const TRAVIS_TRIGGER = 'spinnaker.core.pipeline.config.trigger.travis';
+module(TRAVIS_TRIGGER, [
+  TRAVIS_TRIGGER_OPTIONS_COMPONENT,
+  require('../trigger.directive.js'),
+  IGOR_SERVICE,
+  SERVICE_ACCOUNT_SERVICE,
+  require('../../pipelineConfigProvider.js'),
+]).config((pipelineConfigProvider: any) => {
+  pipelineConfigProvider.registerTrigger({
+    label: 'Travis',
+    description: 'Listens to a Travis job',
+    key: 'travis',
+    controller: 'TravisTriggerCtrl',
+    controllerAs: '$ctrl',
+    templateUrl: require('./travisTrigger.html'),
+    popoverLabelUrl: require('./travisPopoverLabel.html'),
+    manualExecutionHandler: 'travisTriggerExecutionHandler',
+    validators: [
+      {
+        type: 'requiredField',
+        fieldName: 'job',
+        message: '<strong>Job</strong> is a required field on Travis triggers.',
+      },
+      {
+        type: 'serviceAccountAccess',
+        message: `You do not have access to the service account configured in this pipeline's Travis trigger.
+                    You will not be able to save your edits to this pipeline.`,
+        preventSave: true,
+      }
+    ],
+  });
+}).factory('travisTriggerExecutionHandler', ($q: IQService) => {
+  // must provide two fields:
+  //   formatLabel (promise): used to supply the label for selecting a trigger when there are multiple triggers
+  //   selectorTemplate: provides the HTML to show extra fields
+  return {
+    formatLabel: (trigger: IBuildTrigger) => {
+      return $q.when(`(Travis) ${trigger.master}: ${trigger.job}`);
+    },
+    selectorTemplate: '<travis-trigger-options command="$ctrl.command"></travis-trigger-options>',
```

Comment:
@icfantv I've inlined the html. Is the command parameter correct? I couldn't figure out how to test it. It was `vm.command` in the html file.